trade-finance-manager-ui/server/controllers/case/cancellation/cancel-cancellation.controller.post.test.ts

```diff
@@ -0,0 +1,145 @@
+import { createMocks } from 'node-mocks-http';
+import { DEAL_SUBMISSION_TYPE } from '@ukef/dtfs2-common';
+import { aTfmSessionUser } from '../../../../test-helpers';
+import { postCancelCancellation, PostCancelCancellationRequest } from './cancel-cancellation.controller';
+import api from '../../../api';
+
+jest.mock('../../../api', () => ({
+  getDeal: jest.fn(),
+  deleteDealCancellation: jest.fn(),
+}));
+
+const dealId = 'dealId';
+const ukefDealId = 'ukefDealId';
+const mockUser = aTfmSessionUser();
+
+describe('postCancelCancellation', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('redirects to previous page if return=`true`', async () => {
+    // Arrange
+    const previousPage = 'previousPage';
+
+    const { req, res } = createMocks<PostCancelCancellationRequest>({
+      params: { _id: dealId },
+      session: {
+        user: mockUser,
+        userToken: 'a user token',
+      },
+      query: {
+        return: 'true',
+      },
+      body: {
+        previousPage,
+      },
+    });
+
+    // Act
+    await postCancelCancellation(req, res);
+
+    // Assert
+    expect(res._getRedirectUrl()).toBe(previousPage);
+  });
+
+  it('redirects to not found if the deal does not exist', async () => {
+    // Arrange
+    jest.mocked(api.getDeal).mockResolvedValue(undefined);
+
+    const { req, res } = createMocks<PostCancelCancellationRequest>({
+      params: { _id: dealId },
+      session: {
+        user: mockUser,
+        userToken: 'a user token',
+      },
+    });
+
+    // Act
+    await postCancelCancellation(req, res);
+
+    // Assert
+    expect(res._getRedirectUrl()).toBe(`/not-found`);
+  });
+
+  it('redirects to not found if the dealId is invalid', async () => {
+    // Arrange
+    jest.mocked(api.getDeal).mockResolvedValue({ status: 400, data: 'Invalid deal id' });
+
+    const { req, res } = createMocks<PostCancelCancellationRequest>({
+      params: { _id: dealId },
+      session: {
+        user: mockUser,
+        userToken: 'a user token',
+      },
+    });
+
+    // Act
+    await postCancelCancellation(req, res);
+
+    // Assert
+    expect(res._getRedirectUrl()).toBe(`/not-found`);
+  });
+
+  it('redirects to deal summary page if the submission type is invalid (MIA)', async () => {
```

Comment:
Can we add a test to make sure it does not call `api.deleteDealCancellation`?

Reply:
Thanks @MarRobSoftwire looks good, can we just wrap into one describe block though please? Little more DRY e.g

```ts
describe('when the type is invalid (MIA)', () => {
  it('should redirect to deal summary page ...

  it('should not call ...
});
```

Reply:
Have updated :)